### PR TITLE
Don't reset caret when tabbing into a field

### PR DESCRIPTION
**Pull request by JustWalters**
Relates to issues #149 and #211.

### CODE REVIEW

File: lib/number_format.js [hunk 1]
```diff
@@ -949,12 +949,13 @@ var NumberFormat = function (_React$Component) {
...
         var el = e.target;
         var selectionStart = el.selectionStart,
+            selectionEnd = el.selectionEnd,
             _el$value3 = el.value,
             value = _el$value3 === undefined ? '' : _el$value3;
 
 
         var caretPosition = _this4.correctCaretPosition(value, selectionStart);
-        if (caretPosition !== selectionStart) {
+        if (caretPosition !== selectionStart && !(selectionStart === 0 && selectionEnd === value.length)) {
```
Comment (s-yadav) [inline lib/number_format.js]: Can you add a comment on code for `selectionStart === 0 && selectionEnd === value.length` condition.

File: src/number_format.js [hunk 2]
```diff
@@ -795,10 +795,10 @@ class NumberFormat extends React.Component {
     e.persist()
     setTimeout(() => {
       const el = e.target;
-      const {selectionStart, value = ''} = el;
+      const {selectionStart, selectionEnd, value = ''} = el;
 
       const caretPosition = this.correctCaretPosition(value, selectionStart);
-      if (caretPosition !== selectionStart) {
+      if (caretPosition !== selectionStart && !(selectionStart === 0 && selectionEnd === value.length)) {
```
Comment (s-yadav) [inline src/number_format.js]: Can you add a comment on code for `selectionStart === 0 && selectionEnd === value.length` condition.

**Comment by s-yadav** [comment]
Thanks, PR looks good. Will merge this.

**Comment by s-yadav** [comment]
Merging this now. I will add the comment. Thanks.

**Comment by JustWalters** [comment]
Thanks, sorry I didn't get to it.

**Pull request merged**